File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_BuffSpeed.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.Card;
import com.broll.gainea.server.core.objects.Unit;
import com.broll.gainea.server.core.objects.buffs.BuffType;
import com.broll.gainea.server.core.objects.buffs.IntBuff;
import com.broll.gainea.server.core.utils.SelectionUtils;

public class C_BuffSpeed extends Card {


    public C_BuffSpeed() {
        super(28, "Reittier", "Verleiht einer eurer Einheiten eine zusätzliche Bewegungsaktion pro Zug");
    }

    @Override
    public boolean isPlayable() {
        return !owner.getUnits().isEmpty();
    }

    @Override
    protected void play() {
        Unit unit = SelectionUtils.selectPlayerUnit(game, owner, "Wählt eine Einheit");
        IntBuff buff = new IntBuff(BuffType.ADD, 1);
        unit.getMovesPerTurn().addBuff(buff);
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_BuffSpeed.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.objects.buffs.BuffType
import com.broll.gainea.server.core.objects.buffs.IntBuff
import com.broll.gainea.server.core.utils.selectPlayerUnit

class C_BuffSpeed : Card(28, "Reittier", "Verleiht einer eurer Einheiten eine zusätzliche Bewegungsaktion pro Zug") {
    override val isPlayable: Boolean
        get() = owner.units.isNotEmpty()

    override fun play() {
        val unit = game.selectPlayerUnit(owner, "Wählt eine Einheit")
        if (unit != null) {
            val buff = IntBuff(BuffType.ADD, 1)
            unit.movesPerTurn.addBuff(buff)
        }
    }
}
